### PR TITLE
build(ui): split tsconfig into two

### PR DESCRIPTION
**Pull request by scttcper**
#sync-getsentry

jsx files currently can't go to definition or autocomplete because the tsconfig has allowJs set to false and the jsconfig isn't picking up the slack. This change would allow go to definition and find references in all jsx files including tests and story files.

Pros:
- Go to definition and autocomplete in jsx files including storybook, tests
- Remove jsconfig.json

Cons:
- Having two tsconfig files is 2x more complicated than one
- Every place that calls `tsc` needs to call `tsc -p tsconfig.build.json`

### CODE REVIEW

**Comment by billyvg** [comment]
Is there a way to invert it and have build as default  config and configure paths for editors?

**Comment by scttcper** [comment]
@billyvg vscode doesn't support it, but webpack and tsc do ☹️ https://github.com/microsoft/vscode/issues/12463#issuecomment-298149184

**Comment by billyvg** [comment]
@scttcper Can we make a `config` directory -- I've been wanting to split the webpack config too. Maybe something like below: (could also axe `ide`)

```
config/base ---> config/build
            |
            ---> config/ide   ------> root/tsconfig.json
```

**Comment by billyvg** [comment]
@scttcper also note that `getsentry` will also require changes

**Comment by scttcper** [comment]
@billyvg that pattern is supported with typescript 3.9 https://devblogs.microsoft.com/typescript/announcing-typescript-3-9/#solution-style-tsconfig

This is something angular projects have also adopted by default in v10 released yesterday https://github.com/cexbrayat/angular-cli-diff/blob/10.0.0/ponyracer/tsconfig.json

File: config/build/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    "sourceMap": true,
+    "strict": true,
+    "target": "es5",
+    "strictBindCallApply": false,
+    "experimentalDecorators": true,
+    // Skip type checking of all declaration files
+    "skipLibCheck": true,
+    "esModuleInterop": true,
+    "jsx": "preserve",
+    "baseUrl": "../../",
```
Comment (billyvg) [inline config/build/tsconfig.json:33]: hmm, imo, I'd have it as `config/tsconfig.build.json` to remove a level of nested dirs

**Review by billyvg — APPROVED** [review]
LGTM, you will want to merge this at the same time as getsentry, or else things will probably break.

**Pull request merged**